### PR TITLE
Bug fix for issues#240

### PR DESCRIPTION
**Pull request by sweetlikepete**
Added a check to ensure that the cached $desktop_url is equal to the current url before returning a cache hit. I tested to make sure that only subsequent loads of the same page hit the cache and new pages always hit /v1/open. I left in the all the other checks just to be safe.

### CODE REVIEW

**Comment by jsaleigh** [comment]
Hey @sweetlikepete, thanks for this fix!  I'm going to get it into a function in a moment, so that the variables are only set / created in the web build.  Please let me know if you see any issues with it :)

File: src/6_branch.js [hunk 1]
```diff
@@ -376,6 +376,17 @@ Branch.prototype['init'] = wrap(
 			(url ? utils.getParamValue(url) : null);
 		var freshInstall = !sessionData || !sessionData['identity_id'];
 
+		function currentDesktopUrlMatchesSession() {
+			var currentDesktopUrl = window.location.protocol + "//" + window.location.host + window.location.pathname;
```
Comment (dmitrig01) [inline src/6_branch.js]: protocol will have a :?

Comment (dmitrig01) [inline src/6_branch.js]: just checked. guess so. weird

Comment (sweetlikepete) [inline src/6_branch.js]: iknowright

**Pull request merged**